### PR TITLE
Make windowFunnel support other Unsigned Integer Types 

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

Short description (up to few sentences):

Make windowFunnel support other Unsigned Integer Types

Detailed description (optional):

...

### CODE REVIEW

**Comment by sundy-li** [comment]
#5277

**Comment by sundy-li** [comment]
Maybe the first argument of  `sequenceMatch`  should also accept other  Unsigned Integer Types?

**Comment by filimonov** [comment]
Doesn't that change make the binary data incompatible between ch versions?

**Comment by sundy-li** [comment]
> Doesn't that change make the binary data (i.e. serialized function state) incompatible between ch versions?

It's done by c++ template code generation. So the oldest version(UInt32) still keep the same binary 
 function state size.

**Comment by alexey-milovidov** [comment]
> Maybe the first argument of sequenceMatch should also accept other Unsigned Integer Types?

Yes, it will be much better for consistency.

**Comment by alexey-milovidov** [comment]
Do you want to make another PR for sequenceMatch - so I can merge this PR?

**Comment by sundy-li** [comment]
> Do you want to make another PR for sequenceMatch - so I can merge this PR?

Ok, I will make another PR in a few days later.

File: dbms/src/AggregateFunctions/AggregateFunctionWindowFunnel.h [hunk 1]
```diff
@@ -137,34 +135,37 @@ struct AggregateFunctionWindowFunnelData
...
 
-        const_cast<AggregateFunctionWindowFunnelData &>(data).sort();
+        const_cast<Data &>(data).sort();
 
-        // events_timestamp stores the timestamp that latest i-th level event happen withing time window after previous level event.
-        // timestamp defaults to -1, which unsigned timestamp value never meet
-        std::vector<Int32> events_timestamp(events_size, -1);
+        /// events_timestamp stores the timestamp that latest i-th level event happen withing time window after previous level event.
+        /// timestamp defaults to -1, which unsigned timestamp value never meet
+        /// there may be some bugs when UInt64 type timstamp overflows Int64, but it works on most cases.
```
Comment (sundy-li) [inline dbms/src/AggregateFunctions/AggregateFunctionWindowFunnel.h]: @alexey-milovidov  should this limitation be noted at docs?   `timestamp + window`  can't be overflow maximum of Int64

Comment (alexey-milovidov) [reply dbms/src/AggregateFunctions/AggregateFunctionWindowFunnel.h]: Sure.

**Pull request merged**